### PR TITLE
Add Turso as data provider

### PR DESCRIPTION
**Pull request by haaawk**

### CODE REVIEW

**Comment by haaawk** [comment]
@leerob Could you have a look please?

**Comment by overra** [comment]
I'm glad I checked again before starting on this 🙌

File: package.json [hunk 1]
```diff
@@ -22,12 +23,13 @@
...
     "kysely-planetscale": "^1.3.0",
-    "next": "13.2.4",
+    "next": "^13.3.0",
     "postcss": "^8.4.21",
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "tailwindcss": "^3.3.0",
-    "typescript": "^5.0.2"
+    "typescript": "^5.0.2",
+    "vercel": "^28.18.4"
```
Comment (leerob) [inline package.json:32]: I think we can remove this 👍

File: pages/api/turso-global.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { NextRequest as Request, NextResponse as Response } from "next/server";
+import { Client as LibsqlClient, createClient } from "@libsql/client/web";
+
+
```
Comment (leerob) [inline pages/api/turso-global.ts:4]:
```suggestion
```

File: pages/api/turso-global.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  runtime: "edge",
+};
+
+const start = Date.now();
+const client = buildLibsqlClient();
+
+function buildLibsqlClient(): LibsqlClient {
+    const url = process.env.TURSO_DB_URL?.trim();
+    if (url === undefined) {
+        throw new Error("TURSO_DB_URL env var is not defined");
```
Comment (leerob) [inline pages/api/turso-global.ts:15]: You can DM me the credentials on Twitter or lee at vercel dot com.

File: package.json (hunk 1 above)
Comment (haaawk) [reply package.json]: done.

File: pages/api/turso-global.ts (hunk 2 above)
Comment (haaawk) [reply pages/api/turso-global.ts]: done

File: pages/api/turso-global.ts (hunk 3 above)
Comment (haaawk) [reply pages/api/turso-global.ts:14]: Done. Please let me know if you got them.

**Comment by glommer** [comment]
@haaawk we should change the README to add Turso as well, no?

**Comment by haaawk** [comment]
> @haaawk we should change the README to add Turso as well, no?

Done. Thanks for catching that.

**Comment by leerob** [comment]
Could you get the lockfile in sync please?

**Comment by haaawk** [comment]
Thanks @leerob!

**Pull request merged**